pubsub/README.md

```diff
@@ -112,30 +115,34 @@ message Message {
 }
 ```
 
-The `from` field denotes the author of the message, note that this is not
-necessarily the peer who sent the RPC this message is contained in. This is
-done to allow content to be routed through a swarm of pubsubbing peers.
+The `optional` fields may be omitted, depending on the
+[signature policy](#message-signing) and
+[message ID function](#message-identification).
 
-The `data` field is an opaque blob of data, it can contain any data that the
-publisher wants it to.
+The `from` field denotes the author of the message. This is the peer who
+initially authored the message, and NOT the peer who propagated it. Thus, as
+the message is routed through a swarm of pubsubbing peers, the original
+authorship is preserved.
```

Comment:
we should say it is optional.

Reply:
It's already implied by the context. And all fields are optional except for topic ids, which makes this a bit pedantic, but will do.